### ISSUE TITLE
Material ink effects are not clipped in list views

### ISSUE THREAD

**Issue by jpnurmi** [canonical/ubuntu-desktop-installer#52]
Probably a known issue, but the ink effect is rendered outside the border when hovering/pressing a  list tile that has been scrolled partly out:

![image](https://user-images.githubusercontent.com/140617/111704448-6c867380-883f-11eb-95fe-c282d30e4dfb.png)

**Comment by jpnurmi** [comment]
This can be fixed by wrapping the list view with `ClipRRect` to retain rounded corners, and providing a `Material` surface or ink effects inside the clip area:

```
Container(
  decoration: BoxDecoration(
    borderRadius: BorderRadius.circular(40.0),
    border: Border.all(color: Theme.of(context).dividerColor),
  ),
  child: ClipRRect(
    borderRadius: BorderRadius.circular(40.0),
    child: Material(
      child: ListView.builder(
        itemCount: 40,
        itemBuilder: (context, index) {
          return ListTile(
            title: Text(index.toString()),
            onTap: () => {},
          );
        },
      ),
    ),
  ),
)
```

![image](https://user-images.githubusercontent.com/140617/111705285-be7bc900-8840-11eb-85ad-93298f5c42dd.png)

**Comment by jpnurmi** [comment]
I can push a fix proposal after #50 has been merged, to avoid conflicts.

**Comment by oSoMoN** [comment]
Please do, this will be a very welcome visual improvement.
